esp/bindings/SOAP/xpp/fxpp/FragmentedXmlPullParser.h

```diff
@@ -0,0 +1,2145 @@
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#ifndef FXPP_FRAGMENTED_XML_FULL_PARSER_H_
+#define FXPP_FRAGMENTED_XML_FULL_PARSER_H_
+
+#include "xpp/XmlPullParser.h"
+#include <map>
+#include <list>
+#include <set>
+#include <vector>
+
+using namespace xpp;
+
+namespace fxpp
+{
+/**
+ * A collection of interfaces and classes used to parse an XML fragment. A fragment, in this
+ * context, may be a simple, well-formed document that could be parsed by `xpp::XmlPullParser`,
+ * or it may be more complex:
+ *
+ * - A fragment may contain a reference to externally defined markup. Such a reference may be
+ *   detected by the parser and the referenced content is seamlessly injected into the sequence
+ *   of events returned by the parser. The external fragment controls whether it replaces the
+ *   referencing element or descends from it.
+ * - A fragment may contain embedded markup, which is either CDATA-enclosed or XML-encoded XML
+ *   content that a traditional parser would treat as text. Such embedded markup may be detected
+ *   by the parser and the content is seamlessly injected into the sequence of events returned
+ *   by the parser. The embedded fragment controls whether it replaces the parent of the embedded
+ *   text or remains descehded from that element.
+ * - A parser event consumer may define locations within a fragment where external content should
+ *   be injected into the parser's sequence of events. The parser can be compelled to parse such
+ *   content on demand.
+ * - A fragment may be sequence of XML elements with no root element.
+ * - A fragment may be derived from one or more instances of `IPropertyTree`. A node may contain
+ *   text content representing an embedded fragment, or it may contain child elements that must
+ *   be serialized in markup language-specific order. This is not a function of the parser, but
+ *   is a logical extension of the framework.
+ *
+ * This framework attempts to address several complexity concerns introduced by some current and
+ * future configurable components, such as ESDL integration scripts:
+ *
+ * - If a configuration is accessed through a property tree, is the configuration given as a
+ *   property tree element hierarchy or as an element containing text that is the configuration?
+ * - Is it reasonable to assume that the previous question need only be answered once for a
+ *   complete configuration? With configuration reuse encouraged and component reuse happening,
+ *   it does seem likely that the configuration of one component may contain the configuration
+ *   of a second component, defined as embedded markup within.
+ * - If multiple elements in a configuration may contain embedded markup, how many locations
```

Comment:
'locations' meaning different places in the platform code or in plugins to the platform?

Reply:
Places in code. Consider ESDL integration script parsing. The parser gets passed around to all of the operations, which are responsible for parsing themselves. How many operations might have to be changed for each one to handle embedded markup?

This might not be the best example, because I wouldn't recommend many operations be enabled for this, but I expect to have script markup, with possible embedded markup, within markup for other components, which might also have embedded markup.